### PR TITLE
client-go: attach request source with retry info for coprocessor

### LINKED ISSUE #46514: Retry info for request resource of read requests.
## Enhancement

Currently we can't distinguish the retried requests. Because the read requests may be handled not only by leader, the retry info is important for them. We can attach the retry info to the [request soruce](https://github.com/pingcap/kvproto/blob/934e842bfd6ea4b769128570d7814af4f0a07894/proto/kvrpcpb.proto#L852-L854) for read requests.

### PR DESCRIPTION
**Pull request by you06**
### What problem does this PR solve?

Issue Number: close #46514

Problem Summary:

### What is changed and how it works?

Update client-go which will attach the request source with retry info for KV requests. This PR set the required options for coprocessor to enable the extra retry info.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

image

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: session/session.go [hunk 1]
```diff
@@ -4347,9 +4347,8 @@ func (s *session) setRequestSource(ctx context.Context, stmtLabel string, stmtNo
 	if !s.isInternal() {
 		if txn, _ := s.Txn(false); txn != nil && txn.Valid() {
 			txn.SetOption(kv.RequestSourceType, stmtLabel)
-		} else {
-			s.sessionVars.RequestSourceType = stmtLabel
 		}
+		s.sessionVars.RequestSourceType = stmtLabel
```
Comment (you06) [inline session/session.go:4351]: This is a tiny fix because coprocessor will read request source from the session variables.

**Review by ekexium — APPROVED** [review]

**Review by zyguan — APPROVED** [review]

**Pull request merged**